ledger/metrics/collection/summary.js

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env node
+
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+const http = require("http");
+
+const window = "6hours";
+
+const url = target =>
+  `http://localhost:2000/render?target=summarize(${target},%20%27${window}%27,%20%27last%27)&from=-${window}&format=json`;
+
+function fetch(url) {
+  return new Promise((resolve, reject) => {
+    http.get(url, res => {
+      res.setEncoding("utf8");
+      let body = "";
+      res.on("data", data => {
+        body += data;
+      });
+      res.on("end", () => {
+        resolve(JSON.parse(body));
+      });
+      res.on("error", reject);
+    });
+  });
+}
+
+function toDict(rawData) {
+  const result = {};
+  rawData.forEach(row => {
+    const words = row.tags.name.split(".");
+    const target = words.slice(undefined, -1).join(".");
+    const stat = words[words.length - 1];
+    const data = row.datapoints[0][0];
+    result[target] = result[target] || {};
+    result[target][stat] = data;
+  });
+  return result;
+}
+
+const formatNumber = num =>
+  num === undefined || isNaN(num)
+    ? ""
+    : Math.round(num * 100 + Number.EPSILON) / 100;
+
+function toCsv(dictData) {
+  const result = [];
+  for (const key in dictData) {
+    const rowData = dictData[key];
+    const count = formatNumber(rowData.count);
+    const mean = formatNumber(rowData.mean);
+    const total = formatNumber(rowData.mean * rowData.count);
+    const min = formatNumber(rowData.min);
+    const max = formatNumber(rowData.max);
+    result.push(`${key}, ${count}, ${mean}, ${total}, ${min}, ${max}`);
+  }
+  return "Metric, Count, Mean, Total, Min, Max\n" + result.sort().join("\n");
+}
+
+async function fetchData() {
+  const maxDepth = 7;
+  const roots = ["daml", "jvm"];
+
+   // [1, 2, 3, ...]
+  const depths = Array.from(Array(maxDepth), (_, i) => i + 1);
+
+  // ["*", "*.*", "*.*.*", ...]
+  const stars = depths.map(depth =>
+    Array.from(Array(depth), () => "*").join(".")
+  );
+
+  // Array of all URLs to fetch
+  const urls = roots.reduce(
```

Comment:
It'd be nice to have a `flatMap`, right? It's in node.js v12 (as is `flat`), but unfortunately we can't use it here unless we upgrade.

Reply:
Yes, this block is copy pasted from the alternative implementation of `flatMap` from [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap) 😄 

Same for the local method `fetch` - it would be nice to use `node-fetch`, but I did not want to add npm dependencies.